spec/request/service_instances_spec.rb

```diff
@@ -3360,57 +3185,511 @@ def create_bindings(instance, space:, count:)
+      post '/v3/service_instances/some-fake-guid/relationships/shared_spaces', request_body.to_json, space_dev_headers
+
+      expect(last_response).to have_status_code(404)
+      expect(parsed_response['errors']).to include(
+        include(
+          {
+            'detail' => 'Service instance not found',
+            'title' => 'CF-ResourceNotFound'
+          })
+      )
+    end
+
+    describe 'invalid request body' do
+      context 'when request is not a valid relationship' do
+        let(:request_body) do
+          {
+            'data' => { 'guid' => target_space_1.guid }
+          }
+        end
+
+        it 'should respond with 422' do
+          api_call.call(space_dev_headers)
+
+          expect(last_response.status).to eq(422)
+          expect(parsed_response['errors']).to include(
+            include(
+              {
+                'detail' => 'Data must be an array',
+                'title' => 'CF-UnprocessableEntity'
+              })
+          )
+        end
+      end
+
+      context 'when there are additional keys' do
+        let(:request_body) do
+          {
+            'data' => [
+              { 'guid' => target_space_1.guid }
+            ],
+            'fake-key' => 'foo'
+          }
+        end
+
+        it 'should respond with 422' do
+          api_call.call(space_dev_headers)
+
+          expect(last_response.status).to eq(422)
+          expect(parsed_response['errors']).to include(
+            include(
+              {
+                'detail' => "Unknown field(s): 'fake-key'",
+                'title' => 'CF-UnprocessableEntity'
+              })
+          )
+        end
+      end
+    end
+
+    describe 'target space to share to' do
+      context 'does not exist' do
+        let(:target_space_guid) { 'fake-target' }
+        let(:request_body) do
+          {
+            'data' => [
+              { 'guid' => target_space_guid }
+            ]
+          }
+        end
+
+        it 'responds with 422' do
+          api_call.call(space_dev_headers)
+
+          expect(last_response.status).to eq(422)
+          expect(parsed_response['errors']).to include(
+            include(
+              {
+                'detail' => "Unable to share service instance #{service_instance.name} with spaces ['#{target_space_guid}']. Ensure the spaces exist and that you have access to them.",
+                'title' => 'CF-UnprocessableEntity'
+              })
+          )
+        end
+      end
+
+      context 'user does not have access to one of the target spaces' do
+        let(:no_access_target_space) { VCAP::CloudController::Space.make(organization: org) }
+        let(:request_body) do
+          {
+            'data' => [
+              { 'guid' => no_access_target_space.guid },
+              { 'guid' => target_space_1.guid }
+            ]
+          }
+        end
+
+        it 'responds with 422' do
+          api_call.call(space_dev_headers)
```

Comment:
one thing it's not clear from this test is whether the service instance was shared or not with the space the user has access to. Should we add another `it` to cover that (either `it 'does not share with any of the spaces'` or `it 'shares with the other spaces'` or something like that)

Reply:
added an expectation to check that it was not shared at all